cmd/ingest/main_test.go

```diff
@@ -0,0 +1,288 @@
+		AckPolicy: nats.AckExplicitPolicy,
+	})
+	requ.Nil(err)
+
+	t.Cleanup(
+		func() {
+			nc.Close()
+			requ.Nil(natsInstance.Stop())
+			requ.Nil(minioInstance1.Stop())
+			requ.Nil(minioInstance2.Stop())
+		})
+}
+
+func TestRunGroup(t *testing.T) {
+	if v, ok := os.LookupEnv("E2E"); !ok || !(v == "1" || v == "true") {
+		t.Skip("To enable this test, set the E2E environment variable to 1 or true")
+	}
+	setUp(t)
+
+	rawConfig := []byte(fmt.Sprintf(`sources:
+- name: foo_1
+  type: s3
+  endpoint: %s
+  insecure: true
+  bucket: source
+  prefix: prefix/
+  accessKeyID: AKIAIOSFODNN7EXAMPLE
+  secretAccessKey: wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
+- name: foo_2
+  type: s3
+  endpoint: %s
+  insecure: true
+  bucket: source
+  prefix: prefix/
+  accessKeyID: AKIAIOSFODNN7EXAMPLE
+  secretAccessKey: wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
+destinations:
+- name: bar_1
+  type: s3
+  endpoint: %s
+  insecure: true
+  bucket: destination
+  prefix: target_prefix/
+  metafilesPrefix: meta/
+  accessKeyID: AKIAIOSFODNN7EXAMPLE
+  secretAccessKey: wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
+- name: bar_2
+  type: s3
+  endpoint: %s
+  insecure: true
+  bucket: destination
+  prefix: target_prefix/
+  metafilesPrefix: meta/
+  accessKeyID: AKIAIOSFODNN7EXAMPLE
+  secretAccessKey: wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
+workflows:
+- name: foo_1-bar_1
+  source: foo_1
+  destinations:
+  - bar_1
+  interval: 0
+  cleanUp: true
+  webhook: http://localhost:8080 
+- name: foo_2-bar_1-bar_2
+  source: foo_2
+  destinations:
+  - bar_1
+  - bar_2
+  interval: 0
+  webhook: http://localhost:8080 
+`,
+		minioInstance1.Endpoint("minio"),
+		minioInstance2.Endpoint("minio"),
+		minioInstance1.Endpoint("minio"),
+		minioInstance2.Endpoint("minio")))
+
+	c, err := config.New(rawConfig)
+	require.Nil(t, err)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	reg := prometheus.NewRegistry()
+
+	q, err := queue.New(natsInstance.Endpoint("nats"), reg)
+	require.Nil(t, err)
+
+	defer func() {
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+		defer cancel()
+		if err := q.Close(ctx); err != nil {
+			assert.Nil(t, err)
+		}
+	}()
+
+	mc1.MakeBucket(ctx, "source", minio.MakeBucketOptions{})
+	mc2.MakeBucket(ctx, "source", minio.MakeBucketOptions{})
+
+	mc1.MakeBucket(ctx, "destination", minio.MakeBucketOptions{})
+	mc2.MakeBucket(ctx, "destination", minio.MakeBucketOptions{})
+
+	content := "a file"
+	buf := strings.NewReader(content)
+	_, err = mc1.PutObject(ctx, "source", "prefix/file_1", buf, buf.Size(), minio.PutObjectOptions{})
+	require.Nil(t, err)
+	buf = strings.NewReader(content)
+	_, err = mc2.PutObject(ctx, "source", "prefix/file_2", buf, buf.Size(), minio.PutObjectOptions{})
+	require.Nil(t, err)
+	{
+		// enqueue
+		var g run.Group
+		appFlags := &flags{
+			mode:            toPtr(enqueueMode),
+			queueSubject:    toPtr("ingest"),
+			pluginDirectory: toPtr("../../bin/plugin/linux/amd64"),
```

Comment:
Maybe we should template this using the $GOOS and $GOARCH environment variables?